### PR TITLE
[FEAT][TraderApp][Frontend]: Consolidate consignment creation into single-step dialog and fix search race conditions

### PR DESCRIPTION
**Pull request by MohamadNazik**
## Summary

This PR streamlines the consignment creation process by consolidating the multi-step "New Consignment" flow into a single-step dialog. It also resolves a critical UX race condition in the search pickers and improves overall validation feedback and accessibility.

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (no functional changes)

### **Trader App UI Redesign**
- **Unified Creation Modal**: Merged Trade Flow (Import/Export) and CHA selection into a single `NewConsignmentDialog` component.
- **Standalone Component Extraction**: Refactored the dialog logic from `ConsignmentScreen.tsx` into a dedicated internal component for better maintainability.
- **Interaction Enhancements**:
    - Added directional icons (↓/↑) for Trade Flow buttons.
    - Implemented toggle (deselect) behavior for flow selection.
    - Added real-time validation with a red error message and rounded info icon when a CHA is not selected during typing.
    - Optimized modal width to `450px` for a cleaner layout.

### **Bug Fixes & Stability**
- **Search Race Condition**: Resolved the issue where a "double-click" was required for selection in `CHASearch` and `HSCodeSearch`. This was fixed by using `onMouseDown` with `preventDefault()` to capture clicks before focus loss.
- **State Synchronization**: Improved `CHASearch` to clear selections immediately when the search query is modified, while preserving the user's typed input.

### **Accessibility & Quality**
- **ARIA Support**: Added `aria-pressed` and `aria-label` to Trade Flow buttons for better screen reader compatibility.
- **CI Compliance**: Fixed all floating promise warnings and unused imports to ensure clean CI runs.

## Testing

- [x] I have tested this change locally
- [x] I have tested edge cases
- [x] All existing tests pass

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues

- Closes #472 

### 1. Initial Modal View
*Shows the consolidated layout with Trade Flow and CHA search.*
image
### 2. Validation State
*Shows the red error message and rounded icon when a CHA selection is pending during typing.*
image
### 3. Valid Selection
*Shows the active "Create" button once both Flow and CHA are finalized.*
image

## Additional Notes

Merging this PR improves the core trader workflow by reducing cognitive load and interaction friction during the most frequent task: creating new consignments.

### CODE REVIEW

File: portals/apps/trader-app/src/screens/ConsignmentScreen.tsx [hunk 1]
```diff
@@ -14,6 +21,192 @@ import { CHASearch, type CHAOption } from '../components/CHAPicker/CHASearch'
...
+  chaOptions: CHAOption[]
+  creating: boolean
+  onCreate: (data: { flow: TradeFlow; chaId: string }) => Promise<void>
+}
+
+function NewConsignmentDialog({ open, onOpenChange, chaOptions, creating, onCreate }: NewConsignmentDialogProps) {
+  const [newConsignmentData, setNewConsignmentData] = useState({
+    flow: null as TradeFlow | null,
+    chaId: '',
+  })
```
Comment (mushrafmim) [inline portals/apps/trader-app/src/screens/ConsignmentScreen.tsx]: Extract this to a type

File: portals/apps/trader-app/src/screens/ConsignmentScreen.tsx [hunk 2]
```diff
@@ -162,128 +339,14 @@ export function ConsignmentScreen() {
...
-                loading={creating}
-              >
-                {creating ? 'Creating...' : 'Create Consignment'}
-              </Button>
-            ) : null}
-          </Flex>
-        </Dialog.Content>
-      </Dialog.Root>
+      <NewConsignmentDialog
+        key={pickerOpen ? 'open' : 'closed'}
```
Comment (mushrafmim) [inline portals/apps/trader-app/src/screens/ConsignmentScreen.tsx]: key should get a static value. I think not needed since it is not inside a loop.

**Review by mushrafmim — CHANGES_REQUESTED** [review]

**Review by mushrafmim — DISMISSED** [review]

**Review by sthanikan2000 — APPROVED** [review]

**Review by mushrafmim — APPROVED** [review]
